### PR TITLE
Change image values.yaml tree to fit renovate's quirks

### PR DESCRIPTION
**Pull request by kang-makes**
Renovate wants image "spec" to be an object with the parent key called `image`: https://github.com/renovatebot/renovate/blob/1615d262264fd5db4bee76802c3542bc0fd36556/lib/modules/manager/helm-values/util.ts#L5

This should make renovate detect dependencies in the values.

### CODE REVIEW

File: charts/stateless-dns/ci/test-all-possible-values-1.yaml [hunk 1]
```diff
@@ -131,9 +133,11 @@ pdns:
   nameOverride: ""
   fullnameOverride: ""
 
-  registry: ghcr.io
-  repository: txqueuelen/stateless-dns
-  tag: ci  # We are going to test always with the version built or the latest stable
+  image:
+    registry: ghcr.io
+    repository: txqueuelen/stateless-dns
```
Comment (roobre) [inline charts/stateless-dns/ci/test-all-possible-values-1.yaml:138]:
```suggestion
    repository: txqueuelen/stateless-dns/powerdns
```

File: charts/stateless-dns/ci/test-all-possible-values-2.yaml [hunk 2]
```diff
@@ -131,9 +133,11 @@ pdns:
   nameOverride: ""
   fullnameOverride: ""
 
-  registry: ghcr.io
-  repository: txqueuelen/stateless-dns
-  tag: ci  # We are going to test always with the version built or the latest stable
+  image:
+    registry: ghcr.io
+    repository: txqueuelen/stateless-dns
```
Comment (roobre) [inline charts/stateless-dns/ci/test-all-possible-values-2.yaml:138]:
```suggestion
    repository: txqueuelen/stateless-dns/powerdns
```

**Review by roobre — APPROVED** [review]

File: charts/stateless-dns/ci/test-all-possible-values-1.yaml (hunk 1 above)
Comment (kang-makes) [reply charts/stateless-dns/ci/test-all-possible-values-1.yaml:138]: Fair. For the future upgrade path.

File: charts/stateless-dns/ci/test-all-possible-values-2.yaml (hunk 2 above)
Comment (kang-makes) [reply charts/stateless-dns/ci/test-all-possible-values-2.yaml:138]: Fair. For the future upgrade path.

**Pull request merged**